TensorFlow-based DeepDoubleB tagger integration

jmduarte opened the pull request:
This adds the DeepDoubleB TensorFlow based model which is based on a lot of the changes brought in from the DeepFlavour pull request #19893. The [latest BTV meeting presentation](https://indico.cern.ch/event/718578/contributions/2954890/) provides more information about the tagger.

We've checked the compatibility between CMSSW-based discriminators computed using the producers in this PR with the output from the training framework (DeepJet) on a ttbar RelVal sample:

MINIAOD vs reference training framework:
https://cernbox.cern.ch/index.php/s/f8bJcJcS8hfU3WF
![image](https://user-images.githubusercontent.com/4932543/39014280-9103f96e-43df-11e8-80cb-3637bb9ad8dd.png)

RECO vs reference training framework:
https://cernbox.cern.ch/index.php/s/hEqbrrMkz5xBSHs
![image](https://user-images.githubusercontent.com/4932543/39014315-a63d143c-43df-11e8-95e7-5577c3032f9b.png)
The level of discrepancy is similar to what was observed for the DeepFlavour model, as we also train using MINIAOD inputs.

We also checked that our changes don't affect the existing DeepFlavour model:
https://cernbox.cern.ch/index.php/s/KHs8RKuE0qzREq4
![image](https://user-images.githubusercontent.com/4932543/39014251-75606148-43df-11e8-8ffc-089c83a97478.png)
@cvernier @ferencek

Commit:
add double-b features

Commit:
rearrange files

cvernier committed:
update src and interface

kpedro88 committed:
define templates in .h, instantiate in .cc

Commit:
update tensor shape; fix tensor filler

Commit:
fix double-b inputs, except for slight diff in tau1_flightDistance2dSig,tau2_flightDistance2dSig

jmduarte committed:
fix packed candidate loop for AK8 jets (double loop)

jmduarte committed:
add ptratio to charged candidate

jmduarte committed:
make jet radius configurable for drmin

jmduarte committed:
fix producer for RECO/AOD

jmduarte committed:
testing from MINIAOD

jmduarte committed:
remove vertex_associator from deepdoubleb in jetTools

jmduarte committed:
rm printouts and vertex associator

jmduarte committed:
testing configs

jmduarte committed:
update default model file with pt cut and mass sculpting penalty

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-23033/4432

cmsbuild commented:
A new Pull Request was created by  @jmduarte (Javier Duarte) for master.

It involves the following packages:

DataFormats/BTauReco
PhysicsTools/PatAlgos
RecoBTag/DeepFlavour


@perrotta, @monttj, @cmsbuild, @slava77, @gpetruc, @arizzi can you please review it and eventually sign? Thanks.
@TaiSakuma, @gouskos, @imarches, @ahinzmann, @rappoccio, @gpetruc, @mmarionncern, @rovere, @jdamgov, @jdolen, @nhanvtran, @JyothsnaKomaragiri, @gkasieczka, @schoef, @ferencek, @mverzett, @mariadalfonso, @pvmulder, @acaudron, @seemasharmafnal this is something you requested to watch as well.
@davidlange6, @slava77, @fabiocos you are the release manager for this.

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

perrotta commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: 03479b2012be5423c4deb2157b1304dee6f36161
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-23033/27602/summary.html

cmsbuild commented:
Comparison job queued.

slava77 commented:
@jmduarte 
is the new tagger running already with this PR?
I think that it should be enabled if it is planned to be a part of miniAOD (or RECO)

jmduarte committed:
enable DeepDoubleB tagger in MINIAOD

cmsbuild commented:
The code-checks are being triggered in jenkins.

jmduarte commented:
@slava77 with c00e1b3 the tagger should now be enabled for MINIAOD

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-23033/4442

cmsbuild commented:
Pull request #23033 was updated. @perrotta, @monttj, @cmsbuild, @slava77, @gpetruc, @arizzi can you please check and sign again.

slava77 commented:
@cmsbuild please test with cms-sw/cmsdist#3944

cmsbuild commented:
The tests are being triggered in jenkins.
Using externals from cms-sw/cmsdist#3944

cmsbuild commented:
-1

Tested at: c00e1b3fd7facb7f3bcf040cbb6e3330c57533b7

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-23033/27623/summary.html

I found follow errors while testing this PR

**Failed tests:  RelVals**

* **RelVals**:

The relvals timed out after 2 hours.

cmsbuild commented:
Comparison not run due to runTheMatrix errors (RelVals and Igprof tests were also skipped)

slava77 commented:
@cmsbuild please test with cms-sw/cmsdist#3944

cmsbuild commented:
The tests are being triggered in jenkins.
Using externals from cms-sw/cmsdist#3944

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-23033/27628/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 149 differences found in the comparisons
 - DQMHistoTests: Total files compared: 29
 - DQMHistoTests: Total histograms compared: 2492830
 - DQMHistoTests: Total failures: 547
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2492107
 - DQMHistoTests: Total skipped: 176
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 28 files compared)
 - Checked 119 log files, 9 edm output root files, 29 DQM output files

slava77 commented:
just a status update, since we've been silent for a while: the review of the PR is in progress.
First round should be done by tomorrow, hopefully.

slava77 reviewed (commented):
This is a result of the visual inspection. Upcoming tun-time checks may lead to more comments.

- it looks like RecoBTag/DeepFlavour should be renamed: RecoBTag/TensorFlow to avoid using already confusing "DeepFlavour" present in two packages with and without TF
- changes are missing in the JetConverter.h . It was only moved, but the contents were not changed to drop the class "wrapper"

jmduarte committed:
add class versions

jmduarte committed:
add comment that these variables should match the content of DataFormats/BTauReco/interface/TaggingVariable.h

jmduarte committed:
factorize deep b-tagging on top of MINIAOD into applyDeepBtagging_cff.py

jmduarte committed:
remove unnecessary tags in BuildFile.xml; fix indenting

jmduarte committed:
fix naming conventions for converter classes, functions, and include guards

jmduarte committed:
double jetR -> float jetR

jmduarte committed:
add comment on unused variable

jmduarte committed:
remove commented out code

jmduarte committed:
generalize range for different jetR

jmduarte commented on RecoBTag/DeepFlavour/interface/NeutralCandidateConverter.h:
```diff
@@ -0,0 +1,52 @@
+
+
+        n_pf_features.drminsv = catch_infs_and_bound(drminpfcandsv,
+                                                     0,-0.4,0,-0.4);
```
Fixed with 5769437c68b

jmduarte commented on RecoBTag/DeepFlavour/interface/NeutralCandidateConverter.h:
```diff
@@ -0,0 +1,52 @@
+      template <typename CandidateType>
+      static void CommonCandidateToFeatures(const CandidateType * n_pf,
+                                            const reco::Jet & jet,
+                                            const float & drminpfcandsv, const double & jetR,
```
Fixed with 8d1f73f

jmduarte commented on RecoBTag/DeepFlavour/interface/NeutralCandidateConverter.h:
```diff
@@ -0,0 +1,52 @@
+
+      void PackedCandidateToFeatures(const pat::PackedCandidate * n_pf,
+				     const pat::Jet & jet,
+				     const float drminpfcandsv, const double jetR,
```
Fixed with 8d1f73f

jmduarte commented on RecoBTag/DeepFlavour/interface/NeutralCandidateConverter.h:
```diff
@@ -0,0 +1,52 @@
+#ifndef RecoSV_DeepFlavour_NeutralCandidateConverter_h
+#define RecoSV_DeepFlavour_NeutralCandidateConverter_h
```
Fixed with 43c27d3

jmduarte commented on RecoBTag/DeepFlavour/interface/DoubleBTagConverter.h:
```diff
@@ -0,0 +1,20 @@
+namespace btagbtvdeep {
+
+      void DoubleBTagToFeatures(const reco::TaggingVariableList & tag_info_vars,
+                                 BoostedDoubleSVTagInfoFeatures & tag_info_features) ;
```
Fixed with 43c27d3

jmduarte commented on RecoBTag/DeepFlavour/interface/DoubleBTagConverter.h:
```diff
@@ -0,0 +1,20 @@
+#ifndef RecoBTag_DeepFlavour_DoubleBTagConverter_h
+#define RecoBTag_DeepFlavour_DoubleBTagConverter_h
```
Fixed with 43c27d3

jmduarte commented on RecoBTag/DeepFlavour/interface/ChargedCandidateConverter.h:
```diff
@@ -0,0 +1,76 @@
+
+
+  // static data member (avoid undefined ref)
+//  constexpr int ChargedCandidateConverter::qualityMap[8]; 
```
Removed in cc56ffe

jmduarte commented on RecoBTag/DeepFlavour/interface/ChargedCandidateConverter.h:
```diff
@@ -0,0 +1,76 @@
+      void PackedCandidateToFeatures(const pat::PackedCandidate * c_pf,
+                                            const pat::Jet & jet,
+                                            const TrackInfoBuilder & track_info,
+				            const float drminpfcandsv, const double jetR, 
```
Fixed in 8d1f73f

jmduarte commented on RecoBTag/DeepFlavour/interface/ChargedCandidateConverter.h:
```diff
@@ -0,0 +1,76 @@
+
+      }
+    
+      void PackedCandidateToFeatures(const pat::PackedCandidate * c_pf,
```
Fixed in 43c27d3

jmduarte commented on DataFormats/BTauReco/interface/BoostedDoubleSVTagInfoFeatures.h:
```diff
@@ -0,0 +1,42 @@
+namespace btagbtvdeep {
+
+class BoostedDoubleSVTagInfoFeatures {
+
```
Added in 7932110

jmduarte commented on DataFormats/BTauReco/src/classes_def.xml:
```diff
@@ -439,12 +440,15 @@
-  <class name="btagbtvdeep::ChargedCandidateFeatures" ClassVersion="3">
-    <version ClassVersion="3" checksum="1302021141"/>
+  <class name="btagbtvdeep::ChargedCandidateFeatures" ClassVersion="4">
+    <version ClassVersion="4" checksum="2173116472"/>
```
Fixed in 63f21e4

jmduarte commented on DataFormats/BTauReco/src/classes_def.xml:
```diff
@@ -439,12 +440,15 @@
   <class name="btagbtvdeep::DeepFlavourFeatures" ClassVersion="3">
     <version ClassVersion="3" checksum="2487956635"/>
   </class>
+  <class name="btagbtvdeep::DeepDoubleBFeatures"/>
```
Fixed in 63f21e4

jmduarte commented on PhysicsTools/PatAlgos/python/slimming/miniAOD_tools.py:
```diff
@@ -427,6 +427,37 @@ def miniAOD_customizeCommon(process):
+
+    # update slimmed jets to include DeepFlavour (keep same name)
+    # make clone for DeepDoubleB-less slimmed AK8 jets, so output name is preserved
+    process.slimmedJetsAK8NoDeepDoubleB = process.slimmedJetsAK8.clone()
```
Factorized as you recommended in fc938f9

jmduarte commented on RecoBTag/DeepFlavour/BuildFile.xml:
```diff
@@ -1,4 +1,11 @@
 <use name="FWCore/Utilities"/>
+<use name="tensorflow-cc"/>
```
Fixed in c9ddeaa

jmduarte commented on RecoBTag/DeepFlavour/interface/BTagConverter.h:
```diff
@@ -0,0 +1,21 @@
+
+
+      void BTagToFeatures(const reco::TaggingVariableList & tag_info_vars,
+                                 ShallowTagInfoFeatures & tag_info_features);
```
Fixed naming convention in 43c27d3

jmduarte commented on RecoBTag/DeepFlavour/interface/BTagConverter.h:
```diff
@@ -0,0 +1,21 @@
+
+
+      void BTagToFeatures(const reco::TaggingVariableList & tag_info_vars,
+                                 ShallowTagInfoFeatures & tag_info_features);
```
Fixed indentation in 43c27d3

jmduarte commented on RecoBTag/DeepFlavour/interface/ChargedCandidateConverter.h:
```diff
@@ -0,0 +1,76 @@
+#ifndef RecoSV_DeepFlavour_ChargedCandidateConverter_h
+#define RecoSV_DeepFlavour_ChargedCandidateConverter_h
```
Fixed in 43c27d3

jmduarte commented on RecoBTag/DeepFlavour/interface/ChargedCandidateConverter.h:
```diff
@@ -0,0 +1,76 @@
+      // conversion map from quality flags used in PV association and miniAOD one
+      //constexpr static int qualityMap[8]  = {1,0,1,1,4,4,5,6};
+
+      /*enum qualityFlagsShiftsAndMasks {
```
Removed in cc56ffe

jmduarte commented on RecoBTag/DeepFlavour/interface/ChargedCandidateConverter.h:
```diff
@@ -0,0 +1,76 @@
+      void CommonCandidateToFeatures(const CandidateType * c_pf,
+                                            const reco::Jet & jet,
+                                            const TrackInfoBuilder & track_info,
+				            const float & drminpfcandsv, const double & jetR,
```
Fixed indentation in 43c27d3

jmduarte commented on RecoBTag/DeepFlavour/interface/SVConverter.h:
```diff
@@ -0,0 +1,21 @@
+#ifndef RecoBTag_DeepFlavour_SVConverter_h
```
Renamed in 43c27d3

jmduarte commented on RecoBTag/DeepFlavour/interface/SVConverter.h:
```diff
@@ -0,0 +1,21 @@
+
+      void SVToFeatures( const reco::VertexCompositePtrCandidate & sv,
+                                const reco::Vertex & pv, const reco::Jet & jet,
+                                SecondaryVertexFeatures & sv_features) ;
```
Fixed in 43c27d3

jmduarte commented on RecoBTag/DeepFlavour/src/tensor_fillers.cc:
```diff
@@ -42,6 +38,44 @@ namespace btagbtvdeep {
+    *(++ptr) = tag_info_features.trackSip3dSig_2; 
+    *(++ptr) = tag_info_features.trackSip3dSig_3; 
+    *(++ptr) = tag_info_features.z_ratio;
+    //    *(++ptr) = tag_info_features.tau2_vertexDeltaR; 
```
Added comment in bc6103f

jmduarte commented on RecoBTag/DeepFlavour/interface/deep_helpers.h:
```diff
@@ -0,0 +1,71 @@
+
+  // remove infs and NaNs with value  (adapted from DeepNTuples)
+  const float catch_infs(const float in,
+				const float replace_value);
```
I think this split to .h and .cc was done to avoid some issue with circular dependencies when they were inlined... @cvernier @kpedro88 may be able to comment more

kpedro88 replied:
It's been about 6 months since I looked at this. It might have been an ODR violation.

kpedro88 replied:
After some more thought/recollection, I think the problem was this:
* if there is just a .h file and no .cc file, it doesn't become its own translation unit
* if the .h file is then included in multiple other files, it appears in multiple translation units -> ODR violated
* with a .h and .cc file together, it becomes its own translation unit
* that translation unit is then linked to others -> no ODR issue

If *every* function were inlined, splitting to .h and .cc might be avoided. But I'm not sure if this is reasonable (not all functions were inlined in the previous version of this file).

jmduarte commented on RecoBTag/DeepFlavour/interface/tensor_fillers.h:
It is somewhat specific... Although the DeepDoubleC tagger may also use this same set of reduced information

jmduarte commented on RecoBTag/DeepFlavour/src/ChargedCandidateConverter.cc:
```diff
@@ -0,0 +1,90 @@
+
+
+      // instantiate template
+//      template void CommonCandidateToFeatures<reco::PFCandidate>(const reco::PFCandidate, const reco::Jet &, const TrackInfoBuilder &, const float &, ChargedCandidateFeatures &);
```
Removed in 43c27d3afbf0bbc439b916591f2dbea187885bd0

jmduarte commented on RecoBTag/DeepFlavour/src/NeutralCandidateConverter.cc:
```diff
@@ -0,0 +1,40 @@
+      void PackedCandidateToFeatures(const pat::PackedCandidate * n_pf,
+                                            const pat::Jet & jet,
+				     const float drminpfcandsv, const double jetR,
+                                            NeutralCandidateFeatures & n_pf_features) {
```
Checked and fixed in 43c27d3afbf0bbc439b916591f2dbea187885bd0

jmduarte commented on RecoBTag/DeepFlavour/src/JetConverter.cc:
```diff
@@ -0,0 +1,19 @@
+
+
+      static void JetToFeatures(const reco::Jet & jet,
+                                JetFeatures & jet_features) {
```
It was like this before, and it may have caused an issue with ODR violations.

jmduarte committed:
add newline for readability

jmduarte commented on RecoBTag/DeepFlavour/src/deep_helpers.cc:
```diff
@@ -103,8 +67,7 @@ namespace btagbtvdeep {
 
-
+  // instantiate template
+  template bool sv_vertex_comparator<reco::VertexCompositePtrCandidate, reco::Vertex>(const reco::VertexCompositePtrCandidate&, const reco::VertexCompositePtrCandidate&, const reco::Vertex&);
```
added newline in 9bef175

jmduarte committed:
moved unnecessary includes for interface from .h to .cc

jmduarte commented on RecoBTag/DeepFlavour/interface/TrackInfoBuilder.h:
```diff
@@ -0,0 +1,62 @@
+#include "TrackingTools/IPTools/interface/IPTools.h"
+#include "TrackingTools/Records/interface/TransientTrackRecord.h"
+#include "RecoVertex/VertexTools/interface/VertexDistance3D.h"
+#include "TVector3.h"
```
Moved the unnecessary includes for interface from .h to .cc in e6712e351d0

jmduarte committed:
fix indentation in deep_helpers

jmduarte commented on RecoBTag/DeepFlavour/interface/deep_helpers.h:
```diff
@@ -0,0 +1,71 @@
+  const float catch_infs_and_bound(const float in,
+                                          const float replace_value,
+                                          const float lowerbound,
+                                          const float upperbound,
```
Fixed in b090f8e70e1

jmduarte committed:
clang-format DeepDoubleBTagInfoProducer

jmduarte commented on RecoBTag/DeepFlavour/plugins/DeepDoubleBTagInfoProducer.cc:
```diff
@@ -0,0 +1,318 @@
+    typedef reco::VertexCollection VertexCollection;
+    typedef edm::View<reco::BoostedDoubleSVTagInfo> BoostedDoubleSVTagInfoCollection;
+
+	  void beginStream(edm::StreamID) override {}
```
Fixed in 1d7fd03eb0a

jmduarte commented on RecoBTag/DeepFlavour/plugins/DeepDoubleBTagInfoProducer.cc:
```diff
@@ -0,0 +1,318 @@
+    }
+
+    //unsigned int i = 0;
+    //for (const auto * cand : daughters) {	
```
Removed in 1d7fd03eb0a

jmduarte commented on RecoBTag/DeepFlavour/plugins/DeepDoubleBTagInfoProducer.cc:
```diff
@@ -0,0 +1,318 @@
+	if (cand->pt()<0.95) continue;
+	
+	auto packed_cand = dynamic_cast<const pat::PackedCandidate *>(cand);
+	auto reco_cand = dynamic_cast<const reco::PFCandidate *>(cand);
```
Moved down in 43c27d3afbf0bbc439b916591f2dbea187885bd0

jmduarte commented on RecoBTag/DeepFlavour/plugins/DeepDoubleBTagInfoProducer.cc:
```diff
@@ -0,0 +1,318 @@
+	  reco_ptr = reco_ptrs.at(i);
+	} else if (pat_jet && reco_cand) {
+	  reco_ptr = reco_ptrs.at(i);
+	}
```
Changed according to your suggestion in 43c27d3afbf0bbc439b916591f2dbea187885bd0

jmduarte committed:
mv RecoBTag/DeepFlavour -> RecoBTag/TensorFlow

jmduarte committed:
mv RecoBTag/DeepFlavour -> RecoBTag/TensorFlow

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-23033/4634

cmsbuild commented:
Pull request #23033 was updated. @perrotta, @monttj, @cmsbuild, @slava77, @gpetruc, @arizzi can you please check and sign again.

slava77 commented:
@jmduarte 
please post a summary message when you are done with updates to this PR.
Thank you.

perrotta commented:
@cmsbuild please test with cms-sw/cmsdist#3944
(at least, all the points raised in the review seem to be addressed with the latest set of commits: let try to launche the tests, we will re-launch them in case of further updates)

cmsbuild commented:
The tests are being triggered in jenkins.
Using externals from cms-sw/cmsdist#3944

cmsbuild commented:
-1

Tested at: 1bf8db7a2166297c0c368016b5dc8cca4c0b3c0a


The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/1bf8db7a2166297c0c368016b5dc8cca4c0b3c0a
https://github.com/cms-sw/cmssw/commit/3ecec1ebd72ec3d65cfc430b96a769dbb732d9ad
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-23033/27901/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-23033/27901/git-merge-result

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-23033/27901/summary.html

I found follow errors while testing this PR

**Failed tests:  RelVals**

* **RelVals**:

When I ran the RelVals I found an error in the following worklfows:
4.53 step3
<pre>runTheMatrix-results/4.53_RunPhoton2012B+RunPhoton2012B+HLTD+RECODR1reHLT+HARVESTDR1reHLT/step3_RunPhoton2012B+RunPhoton2012B+HLTD+RECODR1reHLT+HARVESTDR1reHLT.log

</pre>136.7611 step2
<pre>runTheMatrix-results/136.7611_RunJetHT2016E_reminiaod+RunJetHT2016E_reminiaod+REMINIAOD_data2016_HIPM+HARVESTDR2_REMINIAOD_data2016_HIPM/step2_RunJetHT2016E_reminiaod+RunJetHT2016E_reminiaod+REMINIAOD_data2016_HIPM+HARVESTDR2_REMINIAOD_data2016_HIPM.log

</pre>136.731 step3
<pre>runTheMatrix-results/136.731_RunSinglePh2016B+RunSinglePh2016B+HLTDR2_2016+RECODR2_2016reHLT_skimSinglePh_HIPM+HARVESTDR2/step3_RunSinglePh2016B+RunSinglePh2016B+HLTDR2_2016+RECODR2_2016reHLT_skimSinglePh_HIPM+HARVESTDR2.log

</pre>136.8311 step2
<pre>runTheMatrix-results/136.8311_RunJetHT2017F_reminiaod+RunJetHT2017F_reminiaod+REMINIAOD_data2017+HARVEST2017_REMINIAOD_data2017/step2_RunJetHT2017F_reminiaod+RunJetHT2017F_reminiaod+REMINIAOD_data2017+HARVEST2017_REMINIAOD_data2017.log

</pre>135.4 step3
<pre>runTheMatrix-results/135.4_ZEE_13+ZEEFS_13+HARVESTUP15FS+MINIAODMCUP15FS/step3_ZEE_13+ZEEFS_13+HARVESTUP15FS+MINIAODMCUP15FS.log

</pre>136.788 step3
<pre>runTheMatrix-results/136.788_RunSinglePh2017B+RunSinglePh2017B+HLTDR2_2017+RECODR2_2017reHLT_skimSinglePh_Prompt+HARVEST2017/step3_RunSinglePh2017B+RunSinglePh2017B+HLTDR2_2017+RECODR2_2017reHLT_skimSinglePh_Prompt+HARVEST2017.log

</pre>9.0 step3
<pre>runTheMatrix-results/9.0_Higgs200ChargedTaus+Higgs200ChargedTaus+DIGI+RECO+HARVEST/step3_Higgs200ChargedTaus+Higgs200ChargedTaus+DIGI+RECO+HARVEST.log

</pre>1306.0 step3
<pre>runTheMatrix-results/1306.0_SingleMuPt1_UP15+SingleMuPt1_UP15+DIGIUP15+RECOUP15+HARVESTUP15/step3_SingleMuPt1_UP15+SingleMuPt1_UP15+DIGIUP15+RECOUP15+HARVESTUP15.log

</pre>25.0 step3
<pre>runTheMatrix-results/25.0_TTbar+TTbar+DIGI+RECOAlCaCalo+HARVEST+ALCATT/step3_TTbar+TTbar+DIGI+RECOAlCaCalo+HARVEST+ALCATT.log

</pre>1330.0 step3
<pre>runTheMatrix-results/1330.0_ZMM_13+ZMM_13+DIGIUP15+RECOUP15_L1TMuDQM+HARVESTUP15_L1TMuDQM/step3_ZMM_13+ZMM_13+DIGIUP15+RECOUP15_L1TMuDQM+HARVESTUP15_L1TMuDQM.log

</pre>10042.0 step3
<pre>runTheMatrix-results/10042.0_ZMM_13+ZMM_13TeV_TuneCUETP8M1_2017_GenSimFull+DigiFull_2017+RecoFull_2017+HARVESTFull_2017+ALCAFull_2017/step3_ZMM_13+ZMM_13TeV_TuneCUETP8M1_2017_GenSimFull+DigiFull_2017+RecoFull_2017+HARVESTFull_2017+ALCAFull_2017.log

</pre>25202.0 step3
<pre>runTheMatrix-results/25202.0_TTbar_13+TTbar_13+DIGIUP15_PU25+RECOUP15_PU25+HARVESTUP15_PU25/step3_TTbar_13+TTbar_13+DIGIUP15_PU25+RECOUP15_PU25+HARVESTUP15_PU25.log

</pre>10024.0 step3
<pre>runTheMatrix-results/10024.0_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2017_GenSimFull+DigiFull_2017+RecoFull_2017+HARVESTFull_2017+ALCAFull_2017/step3_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2017_GenSimFull+DigiFull_2017+RecoFull_2017+HARVESTFull_2017+ALCAFull_2017.log

</pre>11624.0 step3
<pre>runTheMatrix-results/11624.0_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2019_GenSimFull+DigiFull_2019+RecoFull_2019+HARVESTFull_2019+ALCAFull_2019/step3_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2019_GenSimFull+DigiFull_2019+RecoFull_2019+HARVESTFull_2019+ALCAFull_2019.log

</pre>10824.0 step3
<pre>runTheMatrix-results/10824.0_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2018_GenSimFull+DigiFull_2018+RecoFull_2018+HARVESTFull_2018+ALCAFull_2018/step3_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2018_GenSimFull+DigiFull_2018+RecoFull_2018+HARVESTFull_2018+ALCAFull_2018.log

</pre>10224.0 step3
<pre>runTheMatrix-results/10224.0_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2017PU_GenSimFull+DigiFullPU_2017PU+RecoFullPU_2017PU+HARVESTFullPU_2017PU/step3_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2017PU_GenSimFull+DigiFullPU_2017PU+RecoFullPU_2017PU+HARVESTFullPU_2017PU.log

</pre>20034.0 step3
<pre>runTheMatrix-results/20034.0_TTbar_14TeV+TTbar_14TeV_TuneCUETP8M1_2023D17_GenSimHLBeamSpotFull14+DigiFullTrigger_2023D17+RecoFullGlobal_2023D17+HARVESTFullGlobal_2023D17/step3_TTbar_14TeV+TTbar_14TeV_TuneCUETP8M1_2023D17_GenSimHLBeamSpotFull14+DigiFullTrigger_2023D17+RecoFullGlobal_2023D17+HARVESTFullGlobal_2023D17.log

</pre>20434.0 step3
<pre>runTheMatrix-results/20434.0_TTbar_14TeV+TTbar_14TeV_TuneCUETP8M1_2023D19_GenSimHLBeamSpotFull14+DigiFullTrigger_2023D19+RecoFullGlobal_2023D19+HARVESTFullGlobal_2023D19/step3_TTbar_14TeV+TTbar_14TeV_TuneCUETP8M1_2023D19_GenSimHLBeamSpotFull14+DigiFullTrigger_2023D19+RecoFullGlobal_2023D19+HARVESTFullGlobal_2023D19.log

</pre>21234.0 step3
<pre>runTheMatrix-results/21234.0_TTbar_14TeV+TTbar_14TeV_TuneCUETP8M1_2023D21_GenSimHLBeamSpotFull14+DigiFullTrigger_2023D21+RecoFullGlobal_2023D21+HARVESTFullGlobal_2023D21/step3_TTbar_14TeV+TTbar_14TeV_TuneCUETP8M1_2023D21_GenSimHLBeamSpotFull14+DigiFullTrigger_2023D21+RecoFullGlobal_2023D21+HARVESTFullGlobal_2023D21.log

</pre>250202.181 step4
<pre>runTheMatrix-results/250202.181_TTbar_13UP18+TTbar_13UP18+PREMIXUP18_PU25+DIGIPRMXLOCALUP18_PU25+RECOPRMXUP18_PU25+HARVESTUP18_PU25/step4_TTbar_13UP18+TTbar_13UP18+PREMIXUP18_PU25+DIGIPRMXLOCALUP18_PU25+RECOPRMXUP18_PU25+HARVESTUP18_PU25.log

</pre>


The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/1bf8db7a2166297c0c368016b5dc8cca4c0b3c0a
https://github.com/cms-sw/cmssw/commit/3ecec1ebd72ec3d65cfc430b96a769dbb732d9ad
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-23033/27901/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-23033/27901/git-merge-result

cmsbuild commented:
Comparison not run due to runTheMatrix errors (RelVals and Igprof tests were also skipped)

slava77 commented:
```
PhysicsTools/PatAlgos/slimming/miniAOD_tools.py", line 409, in miniAOD_customizeCommon
    updateJetCollection(
NameError: global name 'updateJetCollection' is not defined
```

in case this was working in the local area, I'm guessing that there is a missing commit.

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-23033/4644

cmsbuild commented:
Pull request #23033 was updated. @perrotta, @monttj, @cmsbuild, @slava77, @gpetruc, @arizzi can you please check and sign again.

jmduarte commented:
Due to a merge conflict in `PhysicsTools/PatAlgos/slimming/miniAOD_tools.py` I had to rebase in order to solve the conflict. This was related to the fact that I had factorized the b-tagging out of `PhysicsTools/PatAlgos/slimming/miniAOD_tools.py` into `PhysicsTools/PatAlgos/slimming/applyDeepBtagging_cff.py` while #22396 and #22696 were merged which touched the same file `PhysicsTools/PatAlgos/slimming/miniAOD_tools.py`.

Now the branch should pass tests and have no conflicts.

Regarding @slava77's review, I've addressed all the comments. In general the changes were:
* Rename RecoBTag/DeepFlavour -> RecoBTag/TensorFlow (30b25c0, 4592c89)
* Fixing indentation and other CMS style issues (4ccd865, f89da01, 487b9a9, 9db16a6, 
f57dfd2)
* Keeping single precision on the `jetR` parameter and using it consistently to set variable ranges (5832907, f00f92d)
* Adding class versions in BuildFiles and removing unnecessary tags / formatting (c096d8f, 084c9f7)
* Adding comments and removing commented-out code (1174968, 0f66030, c6da651)
* Factorize b-tagging out of `PhysicsTools/PatAlgos/slimming/miniAOD_tools.py` into `PhysicsTools/PatAlgos/slimming/applyDeepBtagging_cff.py` (3998529)

kpedro88 commented:
please test

cmsbuild commented:
-1

Tested at: 4592c89ddca451b00a9eb57f8f60e6f9dbc4a174

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-23033/27917/summary.html

I found follow errors while testing this PR

**Failed tests:  RelVals**

* **RelVals**:

When I ran the RelVals I found an error in the following worklfows:
4.53 step3
<pre>runTheMatrix-results/4.53_RunPhoton2012B+RunPhoton2012B+HLTD+RECODR1reHLT+HARVESTDR1reHLT/step3_RunPhoton2012B+RunPhoton2012B+HLTD+RECODR1reHLT+HARVESTDR1reHLT.log

</pre>136.7611 step2
<pre>runTheMatrix-results/136.7611_RunJetHT2016E_reminiaod+RunJetHT2016E_reminiaod+REMINIAOD_data2016_HIPM+HARVESTDR2_REMINIAOD_data2016_HIPM/step2_RunJetHT2016E_reminiaod+RunJetHT2016E_reminiaod+REMINIAOD_data2016_HIPM+HARVESTDR2_REMINIAOD_data2016_HIPM.log

</pre>135.4 step3
<pre>runTheMatrix-results/135.4_ZEE_13+ZEEFS_13+HARVESTUP15FS+MINIAODMCUP15FS/step3_ZEE_13+ZEEFS_13+HARVESTUP15FS+MINIAODMCUP15FS.log

</pre>136.8311 step2
<pre>runTheMatrix-results/136.8311_RunJetHT2017F_reminiaod+RunJetHT2017F_reminiaod+REMINIAOD_data2017+HARVEST2017_REMINIAOD_data2017/step2_RunJetHT2017F_reminiaod+RunJetHT2017F_reminiaod+REMINIAOD_data2017+HARVEST2017_REMINIAOD_data2017.log

</pre>136.731 step3
<pre>runTheMatrix-results/136.731_RunSinglePh2016B+RunSinglePh2016B+HLTDR2_2016+RECODR2_2016reHLT_skimSinglePh_HIPM+HARVESTDR2/step3_RunSinglePh2016B+RunSinglePh2016B+HLTDR2_2016+RECODR2_2016reHLT_skimSinglePh_HIPM+HARVESTDR2.log

</pre>9.0 step3
<pre>runTheMatrix-results/9.0_Higgs200ChargedTaus+Higgs200ChargedTaus+DIGI+RECO+HARVEST/step3_Higgs200ChargedTaus+Higgs200ChargedTaus+DIGI+RECO+HARVEST.log

</pre>136.788 step3
<pre>runTheMatrix-results/136.788_RunSinglePh2017B+RunSinglePh2017B+HLTDR2_2017+RECODR2_2017reHLT_skimSinglePh_Prompt+HARVEST2017/step3_RunSinglePh2017B+RunSinglePh2017B+HLTDR2_2017+RECODR2_2017reHLT_skimSinglePh_Prompt+HARVEST2017.log

</pre>25.0 step3
<pre>runTheMatrix-results/25.0_TTbar+TTbar+DIGI+RECOAlCaCalo+HARVEST+ALCATT/step3_TTbar+TTbar+DIGI+RECOAlCaCalo+HARVEST+ALCATT.log

</pre>1306.0 step3
<pre>runTheMatrix-results/1306.0_SingleMuPt1_UP15+SingleMuPt1_UP15+DIGIUP15+RECOUP15+HARVESTUP15/step3_SingleMuPt1_UP15+SingleMuPt1_UP15+DIGIUP15+RECOUP15+HARVESTUP15.log

</pre>1330.0 step3
<pre>runTheMatrix-results/1330.0_ZMM_13+ZMM_13+DIGIUP15+RECOUP15_L1TMuDQM+HARVESTUP15_L1TMuDQM/step3_ZMM_13+ZMM_13+DIGIUP15+RECOUP15_L1TMuDQM+HARVESTUP15_L1TMuDQM.log

</pre>10042.0 step3
<pre>runTheMatrix-results/10042.0_ZMM_13+ZMM_13TeV_TuneCUETP8M1_2017_GenSimFull+DigiFull_2017+RecoFull_2017+HARVESTFull_2017+ALCAFull_2017/step3_ZMM_13+ZMM_13TeV_TuneCUETP8M1_2017_GenSimFull+DigiFull_2017+RecoFull_2017+HARVESTFull_2017+ALCAFull_2017.log

</pre>10024.0 step3
<pre>runTheMatrix-results/10024.0_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2017_GenSimFull+DigiFull_2017+RecoFull_2017+HARVESTFull_2017+ALCAFull_2017/step3_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2017_GenSimFull+DigiFull_2017+RecoFull_2017+HARVESTFull_2017+ALCAFull_2017.log

</pre>25202.0 step3
<pre>runTheMatrix-results/25202.0_TTbar_13+TTbar_13+DIGIUP15_PU25+RECOUP15_PU25+HARVESTUP15_PU25/step3_TTbar_13+TTbar_13+DIGIUP15_PU25+RECOUP15_PU25+HARVESTUP15_PU25.log

</pre>10224.0 step3
<pre>runTheMatrix-results/10224.0_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2017PU_GenSimFull+DigiFullPU_2017PU+RecoFullPU_2017PU+HARVESTFullPU_2017PU/step3_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2017PU_GenSimFull+DigiFullPU_2017PU+RecoFullPU_2017PU+HARVESTFullPU_2017PU.log

</pre>11624.0 step3
<pre>runTheMatrix-results/11624.0_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2019_GenSimFull+DigiFull_2019+RecoFull_2019+HARVESTFull_2019+ALCAFull_2019/step3_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2019_GenSimFull+DigiFull_2019+RecoFull_2019+HARVESTFull_2019+ALCAFull_2019.log

</pre>10824.0 step3
<pre>runTheMatrix-results/10824.0_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2018_GenSimFull+DigiFull_2018+RecoFull_2018+HARVESTFull_2018+ALCAFull_2018/step3_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2018_GenSimFull+DigiFull_2018+RecoFull_2018+HARVESTFull_2018+ALCAFull_2018.log

</pre>20034.0 step3
<pre>runTheMatrix-results/20034.0_TTbar_14TeV+TTbar_14TeV_TuneCUETP8M1_2023D17_GenSimHLBeamSpotFull14+DigiFullTrigger_2023D17+RecoFullGlobal_2023D17+HARVESTFullGlobal_2023D17/step3_TTbar_14TeV+TTbar_14TeV_TuneCUETP8M1_2023D17_GenSimHLBeamSpotFull14+DigiFullTrigger_2023D17+RecoFullGlobal_2023D17+HARVESTFullGlobal_2023D17.log

</pre>21234.0 step3
<pre>runTheMatrix-results/21234.0_TTbar_14TeV+TTbar_14TeV_TuneCUETP8M1_2023D21_GenSimHLBeamSpotFull14+DigiFullTrigger_2023D21+RecoFullGlobal_2023D21+HARVESTFullGlobal_2023D21/step3_TTbar_14TeV+TTbar_14TeV_TuneCUETP8M1_2023D21_GenSimHLBeamSpotFull14+DigiFullTrigger_2023D21+RecoFullGlobal_2023D21+HARVESTFullGlobal_2023D21.log

</pre>20434.0 step3
<pre>runTheMatrix-results/20434.0_TTbar_14TeV+TTbar_14TeV_TuneCUETP8M1_2023D19_GenSimHLBeamSpotFull14+DigiFullTrigger_2023D19+RecoFullGlobal_2023D19+HARVESTFullGlobal_2023D19/step3_TTbar_14TeV+TTbar_14TeV_TuneCUETP8M1_2023D19_GenSimHLBeamSpotFull14+DigiFullTrigger_2023D19+RecoFullGlobal_2023D19+HARVESTFullGlobal_2023D19.log

</pre>250202.181 step4
<pre>runTheMatrix-results/250202.181_TTbar_13UP18+TTbar_13UP18+PREMIXUP18_PU25+DIGIPRMXLOCALUP18_PU25+RECOPRMXUP18_PU25+HARVESTUP18_PU25/step4_TTbar_13UP18+TTbar_13UP18+PREMIXUP18_PU25+DIGIPRMXLOCALUP18_PU25+RECOPRMXUP18_PU25+HARVESTUP18_PU25.log

</pre>

cmsbuild commented:
Comparison not run due to runTheMatrix errors (RelVals and Igprof tests were also skipped)

kpedro88 commented:
please test with cms-sw/cmsdist#3944

cmsbuild commented:
The tests are being triggered in jenkins.
Using externals from cms-sw/cmsdist#3944

cmsbuild commented:
-1

Tested at: 4592c89ddca451b00a9eb57f8f60e6f9dbc4a174

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-23033/27918/summary.html

I found follow errors while testing this PR

**Failed tests:  RelVals**

* **RelVals**:

When I ran the RelVals I found an error in the following worklfows:
136.8311 step2
<pre>runTheMatrix-results/136.8311_RunJetHT2017F_reminiaod+RunJetHT2017F_reminiaod+REMINIAOD_data2017+HARVEST2017_REMINIAOD_data2017/step2_RunJetHT2017F_reminiaod+RunJetHT2017F_reminiaod+REMINIAOD_data2017+HARVEST2017_REMINIAOD_data2017.log

</pre>136.7611 step2
<pre>runTheMatrix-results/136.7611_RunJetHT2016E_reminiaod+RunJetHT2016E_reminiaod+REMINIAOD_data2016_HIPM+HARVESTDR2_REMINIAOD_data2016_HIPM/step2_RunJetHT2016E_reminiaod+RunJetHT2016E_reminiaod+REMINIAOD_data2016_HIPM+HARVESTDR2_REMINIAOD_data2016_HIPM.log

</pre>4.53 step3
<pre>runTheMatrix-results/4.53_RunPhoton2012B+RunPhoton2012B+HLTD+RECODR1reHLT+HARVESTDR1reHLT/step3_RunPhoton2012B+RunPhoton2012B+HLTD+RECODR1reHLT+HARVESTDR1reHLT.log

</pre>135.4 step3
<pre>runTheMatrix-results/135.4_ZEE_13+ZEEFS_13+HARVESTUP15FS+MINIAODMCUP15FS/step3_ZEE_13+ZEEFS_13+HARVESTUP15FS+MINIAODMCUP15FS.log

</pre>136.731 step3
<pre>runTheMatrix-results/136.731_RunSinglePh2016B+RunSinglePh2016B+HLTDR2_2016+RECODR2_2016reHLT_skimSinglePh_HIPM+HARVESTDR2/step3_RunSinglePh2016B+RunSinglePh2016B+HLTDR2_2016+RECODR2_2016reHLT_skimSinglePh_HIPM+HARVESTDR2.log

</pre>136.788 step3
<pre>runTheMatrix-results/136.788_RunSinglePh2017B+RunSinglePh2017B+HLTDR2_2017+RECODR2_2017reHLT_skimSinglePh_Prompt+HARVEST2017/step3_RunSinglePh2017B+RunSinglePh2017B+HLTDR2_2017+RECODR2_2017reHLT_skimSinglePh_Prompt+HARVEST2017.log

</pre>1330.0 step3
<pre>runTheMatrix-results/1330.0_ZMM_13+ZMM_13+DIGIUP15+RECOUP15_L1TMuDQM+HARVESTUP15_L1TMuDQM/step3_ZMM_13+ZMM_13+DIGIUP15+RECOUP15_L1TMuDQM+HARVESTUP15_L1TMuDQM.log

</pre>10042.0 step3
<pre>runTheMatrix-results/10042.0_ZMM_13+ZMM_13TeV_TuneCUETP8M1_2017_GenSimFull+DigiFull_2017+RecoFull_2017/bin/sh:/step3_Higgs200ChargedTaus+Higgs200ChargedTaus+DIGI+RECO+HARVEST.log

</pre>9.0 step3
<pre>runTheMatrix-results/9.0_Higgs200ChargedTaus+Higgs200ChargedTaus+DIGI+RECO+HARVEST/step3_Higgs200ChargedTaus+Higgs200ChargedTaus+DIGI+RECO+HARVEST.log

</pre>25.0 step3
<pre>runTheMatrix-results/25.0_TTbar+TTbar+DIGI+RECOAlCaCalo+HARVEST+ALCATT/step3_TTbar+TTbar+DIGI+RECOAlCaCalo+HARVEST+ALCATT.log

</pre>10024.0 step3
<pre>runTheMatrix-results/10024.0_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2017_GenSimFull+DigiFull_2017+RecoFull_2017+HARVESTFull_2017+ALCAFull_2017/step3_TTbar_13+TTbar_13+DIGIUP15_PU25+RECOUP15_PU25+HARVESTUP15_PU25.log

</pre>25202.0 step3
<pre>runTheMatrix-results/25202.0_TTbar_13+TTbar_13+DIGIUP15_PU25+RECOUP15_PU25+HARVESTUP15_PU25/step3_TTbar_13+TTbar_13+DIGIUP15_PU25+RECOUP15_PU25+HARVESTUP15_PU25.log

</pre>11624.0 step3
<pre>runTheMatrix-results/11624.0_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2019_GenSimFull+DigiFull_2019+RecoFull_2019+HARVESTFull_2019+ALCAFull_2019/step3_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2019_GenSimFull+DigiFull_2019+RecoFull_2019+HARVESTFull_2019+ALCAFull_2019.log

</pre>10824.0 step3
<pre>runTheMatrix-results/10824.0_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2018_GenSimFull+DigiFull_2018+RecoFull_2018+HARVESTFull_2018+ALCAFull_2018/step3_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2018_GenSimFull+DigiFull_2018+RecoFull_2018+HARVESTFull_2018+ALCAFull_2018.log

</pre>10224.0 step3
<pre>runTheMatrix-results/10224.0_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2017PU_GenSimFull+DigiFullPU_2017PU+RecoFullPU_2017PU+HARVESTFullPU_2017PU/step3_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2017PU_GenSimFull+DigiFullPU_2017PU+RecoFullPU_2017PU+HARVESTFullPU_2017PU.log

</pre>21234.0 step3
<pre>runTheMatrix-results/21234.0_TTbar_14TeV+TTbar_14TeV_TuneCUETP8M1_2023D21_GenSimHLBeamSpotFull14+DigiFullTrigger_2023D21+RecoFullGlobal_2023D21+HARVESTFullGlobal_2023D21/step3_TTbar_14TeV+TTbar_14TeV_TuneCUETP8M1_2023D21_GenSimHLBeamSpotFull14+DigiFullTrigger_2023D21+RecoFullGlobal_2023D21+HARVESTFullGlobal_2023D21.log

</pre>20034.0 step3
<pre>runTheMatrix-results/20034.0_TTbar_14TeV+TTbar_14TeV_TuneCUETP8M1_2023D17_GenSimHLBeamSpotFull14+DigiFullTrigger_2023D17+RecoFullGlobal_2023D17+HARVESTFullGlobal_2023D17/step3_TTbar_14TeV+TTbar_14TeV_TuneCUETP8M1_2023D17_GenSimHLBeamSpotFull14+DigiFullTrigger_2023D17+RecoFullGlobal_2023D17+HARVESTFullGlobal_2023D17.log

</pre>20434.0 step3
<pre>runTheMatrix-results/20434.0_TTbar_14TeV+TTbar_14TeV_TuneCUETP8M1_2023D19_GenSimHLBeamSpotFull14+DigiFullTrigger_2023D19+RecoFullGlobal_2023D19+HARVESTFullGlobal_2023D19/step3_TTbar_14TeV+TTbar_14TeV_TuneCUETP8M1_2023D19_GenSimHLBeamSpotFull14+DigiFullTrigger_2023D19+RecoFullGlobal_2023D19+HARVESTFullGlobal_2023D19.log

</pre>250202.181 step4
<pre>runTheMatrix-results/250202.181_TTbar_13UP18+TTbar_13UP18+PREMIXUP18_PU25+DIGIPRMXLOCALUP18_PU25+RECOPRMXUP18_PU25+HARVESTUP18_PU25/step4_TTbar_13UP18+TTbar_13UP18+PREMIXUP18_PU25+DIGIPRMXLOCALUP18_PU25+RECOPRMXUP18_PU25+HARVESTUP18_PU25.log

</pre>

cmsbuild commented:
Comparison not run due to runTheMatrix errors (RelVals and Igprof tests were also skipped)

slava77 commented:
this is now crashing somewhere in DeepDoubleBTagInfoProducer::produce with a segfault.
My guess is for some null ptr dereferencing, but it's just a guess

slava77 commented:
@jmduarte 
please update on the status of this PR

Thank you.

jmduarte committed:
remove pointer to reco candidate; small fix to applyDeepBtagging_cff.py

cmsbuild commented:
The code-checks are being triggered in jenkins.

jmduarte commented:
I updated the PR fixing the segfault locally (at least running `runTheMatrix.py -l 136.8311 -i all` succeeds). It had to do with asking for a pointer to a Reco Candidate from PF Jet, when it was actually a Packed Jet. In the most recent version of the code, this pointer is actually no longer needed so we were able to remove this part of the code altogether.

It would good to test again with cms-sw/cmsdist#3944

Thanks,
Javier

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-23033/4747

cmsbuild commented:
Pull request #23033 was updated. @perrotta, @monttj, @cmsbuild, @slava77, @gpetruc, @arizzi can you please check and sign again.

kpedro88 commented:
please test with cms-sw/cmsdist#3944

cmsbuild commented:
The tests are being triggered in jenkins.
Using externals from cms-sw/cmsdist#3944

cmsbuild commented:
+1
Tested at: 9767f03fbec1f5dbf10c726b65ce545ed877cbd2
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-23033/28066/summary.html

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-23033/28066/summary.html

@slava77 comparisons for the following workflows were not done due to missing matrix map:
   * /build/cmsbld/jenkins/workspace/compare-root-files-short-matrix/results/JR-comparison/PR-23033/136.85_RunEGamma2018A+RunEGamma2018A+HLTDR2_2018+RECODR2_2018reHLT_skimEGamma_Prompt_L1TEgDQM+HARVEST2018_L1TEgDQM

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 154 differences found in the comparisons
 - DQMHistoTests: Total files compared: 31
 - DQMHistoTests: Total histograms compared: 2901712
 - DQMHistoTests: Total failures: 617
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2900905
 - DQMHistoTests: Total skipped: 190
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 30 files compared)
 - Checked 128 log files, 14 edm output root files, 31 DQM output files

kpedro88 commented:
There are changes in the energy and ET of slimmedJetsAK8 in the reco miniAOD comparisons. I'm not sure if this is intended...

jmduarte committed:
reapply Puppi JECs for slimmedAK8 jets after double-b tagging

cmsbuild commented:
The code-checks are being triggered in jenkins.

jmduarte commented:
@kpedro88 Yes, you're right this was unintended.

The issue was (I believe) that we were re-applying the AK8PFchs JECs after b-tagging instead of the AK8PFPuppi JECs (now that the slimmedJetsAK8 collection are PUPPI jets by default). Now I updated this.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-23033/4752

cmsbuild commented:
Pull request #23033 was updated. @perrotta, @monttj, @cmsbuild, @slava77, @gpetruc, @arizzi can you please check and sign again.

kpedro88 commented:
please test with cms-sw/cmsdist#3944

cmsbuild commented:
The tests are being triggered in jenkins.
Using externals from cms-sw/cmsdist#3944

cmsbuild commented:
+1
Tested at: afa6e79ac567a15f66f036e341205432620eaacd
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-23033/28071/summary.html

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-23033/28071/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 67 differences found in the comparisons
 - DQMHistoTests: Total files compared: 31
 - DQMHistoTests: Total histograms compared: 2901712
 - DQMHistoTests: Total failures: 2261
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2899261
 - DQMHistoTests: Total skipped: 190
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 30 files compared)
 - Checked 128 log files, 14 edm output root files, 31 DQM output files

slava77 commented on RecoBTag/TensorFlow/plugins/DeepDoubleBTagInfoProducer.cc:
```diff
@@ -0,0 +1,350 @@
+                if (cand->pt() < 0.95)
+                    continue;
+
+                auto reco_cand = dynamic_cast<const reco::PFCandidate*>(cand);
```
reco_cand is not needed until at least L308. Please move the cast there.

slava77 commented on RecoBTag/TensorFlow/plugins/DeepDoubleBTagInfoProducer.cc:
```diff
@@ -0,0 +1,350 @@
+                                dist = dz;
+                            }
+                        }
+                        auto PV = reco::VertexRef(vtxs, pvi);
```
start local variables and data members with lower case letter. Capitalization is reserved for class names.

slava77 commented on RecoBTag/TensorFlow/src/sorting_modules.cc:
```diff
@@ -1,5 +1,5 @@
 
```
make sure that changes here are matching the changes made in #23335 so that we don't end up with multiple sorting_modules files

slava77 reviewed (commented):
Thank you for the follow up and refactoring of the code. I have several minor comments.

slava77 commented:
@smuzaffar @davidlange6 
do we have control over the detection of possible merge conflicts reported ?

While testing this PR in CMSSW_10_2_X_2018-05-28-2300 I see
```
CONFLICT (rename/rename): Rename "RecoBTag/DeepFlavour/src/sorting_modules.cc"->"RecoBTag/DeepFlavour/plugins/sorting_modules.cc" in branch "HEAD" rename "RecoBTag/DeepFlavour/src/sorting_modules.cc"->"RecoBTag/TensorFlow/src/sorting_modules.cc" in "cms-sw/refs/pull/23033/head"
```

this is somewhat expected to happen due to #23335, but it doesn't show up in the automated merge status checks.

slava77 commented:
@jmduarte 
please make sure to rebase this PR to the latest IB (CMSSW_10_2_X_2018-05-26-2300 or later) when you make updates to this PR.

smuzaffar commented:
@slava77 , no we do not have any control over the dedection of merge conflicts.

jmduarte committed:
add back buildfile

jmduarte committed:
sVToFeatures -> svToFeatures

jmduarte committed:
move reco_cand and rename pv local variable

cmsbuild commented:
The code-checks are being triggered in jenkins.

jmduarte commented:
@slava77, I've made the changes you've requested here: https://github.com/cms-sw/cmssw/pull/23033#pullrequestreview-123788738

In addition, there is only one copy of the `sorting_modules.cc` in `RecoBTag/TensorFlow/plugins` as in #23335.

Finally, we actually had a (non-optimal) bug in our previously trained models, and have submitted a pull request to update the model files here with the same (correct) names: https://github.com/cms-data/RecoBTag-Combined/pull/13

Thanks,
Javier

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-23033/4890

cmsbuild commented:
Pull request #23033 was updated. @perrotta, @monttj, @cmsbuild, @slava77, @gpetruc, @arizzi can you please check and sign again.

slava77 commented:
@cmsbuild please test

IIUC, the name and the format of the files updated in cms-data/RecoBTag-Combined#13 is the same as what's currently in the release. So, the tests should be at least operational.

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
-1

Tested at: 15d011e6632cf444b2f1691ddaa19a7f0f82b3f1

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-23033/28271/summary.html

I found follow errors while testing this PR

**Failed tests:  AddOn**

* **AddOn**:

I found errors in the following addon tests:

cmsDriver.py RelVal -s L1REPACK:Full --data --scenario=pp -n 10 --conditions auto:run2_hlt_Fake2 --relval 9000,50 --datatier "RAW" --eventcontent RAW --customise=HLTrigger/Configuration/CustomConfigs.L1T --era Run2_2016 --fileout file:RelVal_Raw_Fake2_DATA.root --filein /store/data/Run2016B/JetHT/RAW/v1/000/272/762/00000/C666CDE2-E013-E611-B15A-02163E011DBE.root : FAILED - time: date Wed May 30 03:21:44 2018-date Wed May 30 03:15:39 2018 s - exit: 23552
cmsRun /cvmfs/cms-ib.cern.ch/nweek-02526/slc6_amd64_gcc630/cms/cmssw-patch/CMSSW_10_2_X_2018-05-29-1100/src/HLTrigger/Configuration/test/OnLine_HLT_Fake2.py realData=True  globalTag=@ inputFiles=@  : FAILED - time: date Wed May 30 03:21:44 2018-date Wed May 30 03:15:39 2018 s - exit: 21504
cmsDriver.py RelVal -s HLT:Fake2,RAW2DIGI,L1Reco,RECO --data --scenario=pp -n 10 --conditions auto:run2_data_Fake2 --relval 9000,50 --datatier "RAW-HLT-RECO" --eventcontent FEVTDEBUGHLT --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --era Run2_2016 --processName=HLTRECO --filein file:RelVal_Raw_Fake2_DATA.root --fileout file:RelVal_Raw_Fake2_DATA_HLT_RECO.root : FAILED - time: date Wed May 30 03:21:44 2018-date Wed May 30 03:15:39 2018 s - exit: 21504
cmsDriver.py RelVal -s L1REPACK:Full --data --scenario=pp -n 10 --conditions auto:run2_hlt_PRef --relval 9000,50 --datatier "RAW" --customise=HLTrigger/Configuration/CustomConfigs.L1T --era Run2_2018 --eventcontent RAW --fileout file:RelVal_Raw_PRef_DATA.root --filein /store/data/Run2017A/HLTPhysics4/RAW/v1/000/295/606/00000/36DE5E0A-3645-E711-8FA1-02163E01A43B.root : FAILED - time: date Wed May 30 03:22:31 2018-date Wed May 30 03:15:42 2018 s - exit: 23552
cmsRun /cvmfs/cms-ib.cern.ch/nweek-02526/slc6_amd64_gcc630/cms/cmssw-patch/CMSSW_10_2_X_2018-05-29-1100/src/HLTrigger/Configuration/test/OnLine_HLT_PRef.py realData=True  globalTag=@ inputFiles=@  : FAILED - time: date Wed May 30 03:22:31 2018-date Wed May 30 03:15:42 2018 s - exit: 21504
cmsDriver.py RelVal -s HLT:PRef,RAW2DIGI,L1Reco,RECO --data --scenario=pp -n 10 --conditions auto:run2_data_PRef --relval 9000,50 --datatier "RAW-HLT-RECO" --eventcontent FEVTDEBUGHLT --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --era Run2_2018 --processName=HLTRECO --filein file:RelVal_Raw_PRef_DATA.root --fileout file:RelVal_Raw_PRef_DATA_HLT_RECO.root : FAILED - time: date Wed May 30 03:22:31 2018-date Wed May 30 03:15:42 2018 s - exit: 21504
cmsDriver.py RelVal -s L1REPACK:Full2015Data --data --scenario=HeavyIons -n 10 --conditions auto:run2_hlt_HIon --relval 9000,50 --datatier "RAW" --eventcontent RAW --customise=HLTrigger/Configuration/CustomConfigs.L1T --era Run2_2016,Run2_HI --fileout file:RelVal_Raw_HIon_DATA.root --filein /store/hidata/HIRun2015/HIHardProbes/RAW-RECO/HighPtJet-PromptReco-v1/000/263/689/00000/1802CD9A-DDB8-E511-9CF9-02163E0138CA.root : FAILED - time: date Wed May 30 03:19:58 2018-date Wed May 30 03:15:44 2018 s - exit: 23552
cmsRun /cvmfs/cms-ib.cern.ch/nweek-02526/slc6_amd64_gcc630/cms/cmssw-patch/CMSSW_10_2_X_2018-05-29-1100/src/HLTrigger/Configuration/test/OnLine_HLT_HIon.py realData=True  globalTag=@ inputFiles=@  : FAILED - time: date Wed May 30 03:19:58 2018-date Wed May 30 03:15:44 2018 s - exit: 21504
cmsDriver.py RelVal -s HLT:HIon,RAW2DIGI,L1Reco,RECO --data --scenario=HeavyIons -n 10 --conditions auto:run2_data_HIon --relval 9000,50 --datatier "RAW-HLT-RECO" --eventcontent FEVTDEBUGHLT --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --era Run2_2016,Run2_HI --processName=HLTRECO --filein file:RelVal_Raw_HIon_DATA.root --fileout file:RelVal_Raw_HIon_DATA_HLT_RECO.root : FAILED - time: date Wed May 30 03:19:58 2018-date Wed May 30 03:15:44 2018 s - exit: 21504
cmsDriver.py RelVal -s L1REPACK:Full --data --scenario=pp -n 10 --conditions auto:run2_hlt_PIon --relval 9000,50 --datatier "RAW" --customise=HLTrigger/Configuration/CustomConfigs.L1T --era Run2_2018 --eventcontent RAW --fileout file:RelVal_Raw_PIon_DATA.root --filein /store/data/Run2017A/HLTPhysics4/RAW/v1/000/295/606/00000/36DE5E0A-3645-E711-8FA1-02163E01A43B.root : FAILED - time: date Wed May 30 03:24:16 2018-date Wed May 30 03:19:52 2018 s - exit: 23552
cmsRun /cvmfs/cms-ib.cern.ch/nweek-02526/slc6_amd64_gcc630/cms/cmssw-patch/CMSSW_10_2_X_2018-05-29-1100/src/HLTrigger/Configuration/test/OnLine_HLT_PIon.py realData=True  globalTag=@ inputFiles=@  : FAILED - time: date Wed May 30 03:24:16 2018-date Wed May 30 03:19:52 2018 s - exit: 21504
cmsDriver.py RelVal -s HLT:PIon,RAW2DIGI,L1Reco,RECO --data --scenario=pp -n 10 --conditions auto:run2_data_PIon --relval 9000,50 --datatier "RAW-HLT-RECO" --eventcontent FEVTDEBUGHLT --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --era Run2_2018 --processName=HLTRECO --filein file:RelVal_Raw_PIon_DATA.root --fileout file:RelVal_Raw_PIon_DATA_HLT_RECO.root : FAILED - time: date Wed May 30 03:24:16 2018-date Wed May 30 03:19:52 2018 s - exit: 21504
cmsDriver.py RelVal -s L1REPACK:GT1   --data --scenario=pp -n 10 --conditions auto:run1_hlt_Fake --relval 9000,50 --datatier "RAW" --eventcontent RAW --customise=HLTrigger/Configuration/CustomConfigs.L1T --fileout file:RelVal_Raw_Fake_DATA.root --filein /store/data/Run2012A/MuEG/RAW/v1/000/191/718/14932935-E289-E111-830C-5404A6388697.root : FAILED - time: date Wed May 30 03:27:06 2018-date Wed May 30 03:21:50 2018 s - exit: 23552
cmsRun /cvmfs/cms-ib.cern.ch/nweek-02526/slc6_amd64_gcc630/cms/cmssw-patch/CMSSW_10_2_X_2018-05-29-1100/src/HLTrigger/Configuration/test/OnLine_HLT_Fake.py realData=True  globalTag=@ inputFiles=@  : FAILED - time: date Wed May 30 03:27:06 2018-date Wed May 30 03:21:50 2018 s - exit: 21504
cmsDriver.py RelVal -s HLT:Fake,RAW2DIGI,L1Reco,RECO --data --scenario=pp -n 10 --conditions auto:run1_data_Fake --relval 9000,50 --datatier "RAW-HLT-RECO" --eventcontent FEVTDEBUGHLT --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --processName=HLTRECO --filein file:RelVal_Raw_Fake_DATA.root --fileout file:RelVal_Raw_Fake_DATA_HLT_RECO.root : FAILED - time: date Wed May 30 03:27:06 2018-date Wed May 30 03:21:50 2018 s - exit: 21504
cmsRun /cvmfs/cms-ib.cern.ch/nweek-02526/slc6_amd64_gcc630/cms/cmssw-patch/CMSSW_10_2_X_2018-05-29-1100/src/Utilities/ReleaseScripts/scripts/read312RV_cfg.py : FAILED - time: date Wed May 30 03:28:34 2018-date Wed May 30 03:24:18 2018 s - exit: 23552

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-23033/28271/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 61 differences found in the comparisons
 - DQMHistoTests: Total files compared: 31
 - DQMHistoTests: Total histograms compared: 2902471
 - DQMHistoTests: Total failures: 16
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2902265
 - DQMHistoTests: Total skipped: 190
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 30 files compared)
 - Checked 128 log files, 14 edm output root files, 31 DQM output files

cmsbuild commented:
The tests are being triggered in jenkins.

slava77 commented on RecoBTag/TensorFlow/plugins/sorting_modules.cc:
```diff
@@ -1,5 +1,5 @@
 
```
this should have been moved to src/, now that all other tools were mover to src/ and exposed in interface/

cmsbuild commented:
+1
Tested at: 15d011e6632cf444b2f1691ddaa19a7f0f82b3f1
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-23033/28299/summary.html

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-23033/28299/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 61 differences found in the comparisons
 - DQMHistoTests: Total files compared: 31
 - DQMHistoTests: Total histograms compared: 2902471
 - DQMHistoTests: Total failures: 17
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2902264
 - DQMHistoTests: Total skipped: 190
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 30 files compared)
 - Checked 128 log files, 14 edm output root files, 31 DQM output files

slava77 commented:
@jmduarte 
please clarify if you can update the RecoBTag/TensorFlow/plugins/sorting_modules.cc location soon.
Thank you.

jmduarte committed:
mv sorting_modules

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-23033/4934

cmsbuild commented:
Pull request #23033 was updated. @perrotta, @monttj, @cmsbuild, @slava77, @gpetruc, @arizzi can you please check and sign again.

slava77 commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: 2f773313d8e55ef8d6ecf4082b89e380825a97de
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-23033/28326/summary.html

cmsbuild commented:
Comparison job queued.

slava77 commented:
@cmsbuild please test with cms-sw/cmsdist#4063

cmsbuild commented:
The tests are being triggered in jenkins.
Using externals from cms-sw/cmsdist#4063

slava77 commented:
+1

for https://github.com/cms-sw/cmssw/pull/23033	2f77331
- changes are in line with the PR descriptions and the follow up review. 
- jenkins tests pass and comparisons with the baseline show differences in the miniAOD slimmed AK8 jets:
    - 2 new b-tag discriminants appear, which is fully expected
    - float-level precision differences show up in AK8 JetMET comparisons, which is a downside of the jet update step (which includes re-application of jet correction)
- local tests with 1K events on 1325.51 (ttbar) and 136.8311 (JetHT 2017F data) show expected behavior:
    - memory use increases by about 6MB; 5MB of these scale by the number of streams (3 MB in the TF session, and 2 MB in the TF inference data per event), the remaining 1MB is in the global cache of the TF graph
    - CPU time from 10 added modules is dominated by pfDeepDoubleBJetTagsSlimmedAK8DeepDoubleB (DeepDoubleBTFJetTagsProducer); the total increase is about 2% of the miniAOD processing time.
    - tests with the older training show only somewhat meaningful distributions for the discriminants:
probH distribution in ttbar has a more significant tail at higher values 
![all_origvssign1017_ttbar13reminiaodwf1325p51c_min2 max-2 patjets_slimmedjetsak8__pat_obj___pairdiscrivector__8__second](https://user-images.githubusercontent.com/4676718/40800504-6c11eaec-64c5-11e8-8dc0-48526a4c5d00.png)
compared to the JetHT data
![all_origvssign1017_runjetht2017freminiaod136p8311c_min2 max-2 patjets_slimmedjetsak8__pat_obj___pairdiscrivector__8__second](https://user-images.githubusercontent.com/4676718/40800520-776bb13e-64c5-11e8-8498-c013017daf20.png)

After  cms-sw/cmsdist#4063 the discrimination is much more clear (current IB+this PR is in black and IB+this PR+cms-sw/cmsdist#4063 is in red):
in ttbar probH
![all_sign1017-pass-data-58ae0a8vssign1017_ttbar13reminiaodwf1325p51c_min2 max-2 patjets_slimmedjetsak8__pat_obj___pairdiscrivector__8__second](https://user-images.githubusercontent.com/4676718/40803617-eb6e578c-64cd-11e8-8bed-57904fc2fed3.png)
probQ
![all_sign1017-pass-data-58ae0a8vssign1017_ttbar13reminiaodwf1325p51c_min2 max-2 patjets_slimmedjetsak8__pat_obj___pairdiscrivector__9__second](https://user-images.githubusercontent.com/4676718/40803638-f8d57310-64cd-11e8-82c3-eb0765caa46a.png)


The update in cms-sw/cmsdist#4063 can go in independently.

fabiocos commented:
@jmduarte @slava77 while checking this PR for integration, I saw that there is a list of warnings during the compilation. Most of them come from the external Eigen, but the first one seems tpo be specific of the CMSSW code touched in this PR, it would be better to fix is before merging

@smuzaffar FYI

slava77 commented on RecoBTag/TensorFlow/src/JetConverter.cc:
```diff
@@ -1,25 +1,19 @@
-                                JetFeatures & jet_features) {
+  static void jetToFeatures(const reco::Jet & jet,
+			    JetFeatures & jet_features) {
         jet_features.pt = jet.pt(); // uncorrected
```
this function is already defined in JetConverter.h and the compiler generates a warning

slava77 replied:
I think that either the implementation needs to be removed in .h, or this .cc file should just simply include .h. The latter seems better to me.

jmduarte replied:
@slava77 thanks for the suggestion. So it's ok to have `JetConverter.cc` be just one line?
```
#include "DataFormats/BTauReco/interface/JetFeatures.h"
```
thanks

slava77 replied:
yes, AFAIK this works

slava77 commented:
On 6/1/18 2:05 AM, Fabio Cossutti wrote:
> @jmduarte <https://github.com/jmduarte> @slava77
> <https://github.com/slava77> while checking this PR for integration, I
> saw that there is a list of warnings during the compilation. Most of
> them come from the external Eigen, but the first one seems tpo be
> specific of the CMSSW code touched in this PR, it would be better to fix
> is before merging
> 

I suppose that this is about

RecoBTag/TensorFlow/src/JetConverter.cc:9:15: warning: 'void
btagbtvdeep::jetToFeatures(const reco::Jet&, btagbtvdeep::JetFeatures&)'
defined but not used [-Wunused-function]
   static void jetToFeatures(const reco::Jet & jet,

I agree, it makes sense to fix.


> @smuzaffar <https://github.com/smuzaffar> FYI
> 

It would be nice to get the bot summaries to pick up the new warnings.

jmduarte committed:
remove multiple definition of jetToFeatures

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-23033/4959

cmsbuild commented:
Pull request #23033 was updated. @perrotta, @monttj, @cmsbuild, @slava77, @gpetruc, @arizzi can you please check and sign again.

slava77 commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: a3fceea54fff910b7133e95ad09365a8f6329bb5
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-23033/28372/summary.html

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-23033/28372/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 61 differences found in the comparisons
 - DQMHistoTests: Total files compared: 31
 - DQMHistoTests: Total histograms compared: 2902471
 - DQMHistoTests: Total failures: 17
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2902264
 - DQMHistoTests: Total skipped: 190
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 30 files compared)
 - Checked 128 log files, 14 edm output root files, 31 DQM output files

fabiocos commented:
+1

fabiocos commented:
merge

jmduarte commented:
Hi @slava77, for submitting a backport to 94X as discussed previously, do we want to keep all the changes (including renameing RecoBTag/DeepFlavour -> RecoBTag/TensorFlow, etc.)?

One key difference we will implement is we will disable this in MINIAOD by default (so the code will be there for users to recompute in 94X, but it will not be produced in MINIAOD by default).

Thanks,
Javier

slava77 commented:
On 6/6/18 9:44 AM, Javier Duarte wrote:
> Hi @slava77 <https://github.com/slava77>, for submitting a backport to
> 94X as discussed previously, do we want to keep all the changes
> (including renameing RecoBTag/DeepFlavour -> RecoBTag/TensorFlow, etc.)?
> 
> One key difference we will implement is we will disable this in MINIAOD
> by default (so the code will be there for users to recompute in 94X, but
> it will not be produced in MINIAOD by default).
> 
> Thanks,
> Javier
> 

Hi Javier,
I find it most practical to have a full backport, including the file
renaming.

Will the double-B expected to be used in 94X only for analysis?
